### ISSUE TITLE
python imports

### ISSUE THREAD

**Issue by shindavid** [shindavid/AlphaZeroArcade#8]
### **The Problem**

One of my biggest complaints about python is this problem: https://stackoverflow.com/questions/714063/importing-modules-from-parent-folder

There are a bunch of solutions offered in the answers, but none of them are perfect. It's frankly embarrassing for a language like python to have this sort of problem still.

In any case, in our current project, we have this same problem. The solution I chose is `sys.path` hackery. For instance, in `py/connect4/game_logic.py`, we have:

```
sys.path.append(os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
```

...which allows the subsequent imports to work:

```
from interface import AbstractGameState, ActionIndex, ActionMask, GameResult
```

**The downside of this is that my IDE's refactoring tools become quite fragile**. When I move a block of code from one file to another, it doesn't know to add the `sys.path` hack, or to place moved imports below an existing `sys.path` hack line.

### **Proposed Solution**

When I worked at Jump, I devised a custom solution to this problem. Essentially, I created a wrapper around `python`. It was called `spython` ("s" for "stormfx", the name of our team). This `spython` executable was simply a python file that roughly looked something like this:

```
#!/usr/bin/env python 
...
repo_root = deduce_repo_root_from_cwd()
if repo_root:
  py_dir = os.path.join(repo_root, 'py')
  append_to_PYTHONPATH_env_var(py_dir)
sys.exit(os.system(f'python {forward_args(sys.args)}'))
```

Basically, if we were anywhere inside of a repo checkout, it would automatically add `$REPO_ROOT/py/` to `$PYTHONPATH` and then dispatch to `python`. Determination of `$REPO_ROOT` was accomplished by searching for the `REPO_ROOT_MARKER` file in the cwd and its ancestor directories.

Then, from within a repo checkout, instead of doing `python file.py`, you would just do `spython file.py`.

Of course, for this to work, you need to have a copy of this `spython` file, and it needs to be in your `$PATH`. For this purpose, we had a special `box_setup.py` script in the repo, which would copy the file from the local repo checkout to the `~/repo_bin/` directory, and modify your `.bashrc` to add that directory to your `$PATH`.

I proposed we copy this solution in our project.
